src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldDescription.hs

```diff
@@ -18,7 +18,7 @@ import Swarm.Game.Entity
 import Swarm.Game.Land
 import Swarm.Game.Location
 import Swarm.Game.Scenario.RobotLookup
-import Swarm.Game.Scenario.Topography.Area (emptyGrid)
+import Swarm.Game.Scenario.Topography.Area (Grid (EmptyGrid))
```

Comment:
It seems like a bit of a code smell to import a constructor of `Grid` just to be able to make an empty grid, especially if the eventual goal is to make a smart constructor.  I think the constructors of `Grid` should not be exported at all.  That way this module (and others like it) won't have to be updated if the constructors ever change again. We could either (1) continue to export `emptyGrid` as before, or (2) (slightly fancier) make an instance of `AsEmpty` and then import `pattern Empty` from `Control.Lens.Empty`.
```
instance AsEmpty Grid where
  _Empty = prism' (const EmptyGrid) (\case {EmptyGrid -> Just (); _ -> Nothing})
```

Reply:
For now there are still a number of places where the `Grid` constructor is used.  Hiding exports of the constructors will be a prerequisite for completing #1981.